### ISSUE TITLE
Support type=git buildpack registry

### ISSUE THREAD

**Issue by jromero** [buildpacks/pack#746]
### Description

As a buildpack author, I'd like to distribute my buildpacks using a non-GitHub registry. For example, GitLab, Bitbucket or vanilla git repo.

### Proposed solution

As proposed in [RFC #44](https://github.com/buildpacks/rfcs/blob/main/text/0044-pack-publish-buildpack.md) a [`git` type](https://github.com/buildpacks/rfcs/blob/main/text/0044-pack-publish-buildpack.md#type-git-schema) would be available.

```toml
[[registries]]
name="<string>"
type="git"
url="<path to git repo>"
```

This type would create a commit message directly on a clone of the repo:

Format:
```
[<ADD|YANK>] <namespace>/<name>@<version>

<user message (optional)>
```

... and pushes it to remote registry. For more details see [this section](https://github.com/buildpacks/rfcs/blob/main/text/0044-pack-publish-buildpack.md#pack-register-buildpack-url) in the RFC.

### Additional context
- [ ] This feature should be documented somewhere

**Comment by jromero** [comment]
@supra08 will be taking a look at this. I will flesh out more details.

**Comment by jromero** [comment]
@supra08 curious to know if there are any update on this?

**Comment by supra08** [comment]
Yeah working on this. Things got delayed a bit due to some other work but this will be in good shape in a few days.

**Comment by elbandito** [comment]
> @supra08 will be taking a look at this. I will flesh out more details.

Thanks for your help on this @supra08!  Feel free to reach out to me with any help/support.

**Comment by jromero** [comment]
Resolved by #782 (#819)
